src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ConditionalWcfTest.cs

```diff
@@ -10,42 +14,198 @@ namespace Infrastructure.Common
+
+                bool evaluatedResult = false;
+
+                // Highest precedence: environment variable if set and can be parsed
+                string value = Environment.GetEnvironmentVariable(conditionName);
+                bool parsedValue = false;
+                if (!String.IsNullOrWhiteSpace(value) && bool.TryParse(value, out parsedValue))
+                {
+                    result = parsedValue;
+                    evaluatedResult = true;
+                }
+
+                // Next precedence: TestProperties if present and can be parsed
+                else if (TestProperties.PropertyNames.Contains(conditionName))
+                {
+                    value = TestProperties.GetProperty(conditionName);
+                    if (!String.IsNullOrWhiteSpace(value) && bool.TryParse(value, out parsedValue))
+                    {
+                        result = parsedValue;
+                        evaluatedResult = true;
+                    }
+                }
+
+                // Next precedence: optional runtime detection func
+                if (!evaluatedResult && detectFunc != null)
+                {
+                    result = detectFunc();
+                    evaluatedResult = true;
+                }
+
+                // Final precedence: false
+                else if (!evaluatedResult && detectFunc != null)
+                {
+                    result = false;
+                }
+
+                _evaluatedConditions[conditionName] = result;
+                return result;
+            }
+        }
+
+        private static bool Is_Server_Local()
         {
-            string propertyValue = TestProperties.GetProperty(testCategoryName);
-            bool isEnabled;
-            if (!String.IsNullOrEmpty(propertyValue) && bool.TryParse(propertyValue, out isEnabled))
+            // Temporary workaround to detect whether this test is using localhost
+            // for either new or old test infrastructure.
+            // Refactor this after integration to address https://github.com/dotnet/wcf/issues/1024 
+            // to use centralized helper method.
+            string host = null;
+            if (TestProperties.PropertyNames.Contains("ServiceUri"))
+            {
+                host = TestProperties.GetProperty("ServiceUri");
+            }
+            else if (TestProperties.PropertyNames.Contains("BridgeHost"))
             {
-                return isEnabled;
+                host = TestProperties.GetProperty("BridgeHost");
             }
 
+            if (String.IsNullOrWhiteSpace(host))
+            {
+                return false;
+            }
+
+            int index = host.IndexOf("localhost", 0, StringComparison.OrdinalIgnoreCase);
+            return index >= 0;
+        }
+        
+        private static bool Is_Windows()
+        {
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+        }
+
+        private static bool Is_NET_Native()
```

Comment:
Not seen this is used anywhere. Maybe we can include it when we actually use it?